eng/common/TestResources/deploy-test-resources.yml

```diff
@@ -27,13 +42,31 @@ steps:
         ${{ parameters.SubscriptionConfiguration }}
       "@ | ConvertFrom-Json -AsHashtable;
 
+      if ($subscriptionConfiguration.ArmTemplateParameters) {
+        # Handle subscription configuration containing arm template parameters
+        $armTemplateParameters = $subscriptionConfiguration.ArmTemplateParameters
```

Comment:
Instead of blindly cutting over couldn't we set the ArmTemplateParameters yml parameter's default value to be `$subscriptionConfiguration.ArmTemplateParameters` and then we go and remove any places that pass a value and test it as you remove them? 

Right now as it stands this will ignore any parameters.ArmTemplateParameters everywhere as soon as you put it in KV and so there will be dead parameters and we wouldn't have even tested the change before we changed this under everyone.

Reply:
Yep, good catch, the default logic should have picked the local string. I can change it to default parameters.